external/loaders/loaders/_utils.py

```diff
@@ -71,6 +71,18 @@ def stack(unstacked_dims: Sequence[str], ds: xr.Dataset) -> xr.Dataset:
     return ds_stacked.transpose(SAMPLE_DIM_NAME, *unstacked_dims)
 
 
+@curry
+def select_random_ordered_subset(fraction: float, ds: xr.Dataset) -> xr.Dataset:
```

Comment:
This function is pretty short, but does 2-3 things and could be hard to test: computes `n_samples`, . shuffle, orders, selects, and it has a hard coded dependence on `SAMPLE_DIM_NAME` that will make future refactors more difficult. I would try to make these more composable. Would something like this be equally performant?
```
batches.map(shuffle).map(select_fraction(fraction)).map(sort_by_time)
```

Reply:
@nbren12 we can't `shuffle` at the start because the random subset must have its indices in-order for dask access. Shuffle-then-order would break performance. tfdataset might do something smarter here, but our sequences code won't.

This could use a test. I'd suggest selecting from an artificially-constructed dataset where each sample has a specific value, then asserting that the result has 1) the correct number of samples, 2) no repeated sample values, and 3) monotonically-increasing sample values.

Reply:
> @nbren12 we can't shuffle at the start because the random subset must have its indices in-order for dask access.

I see, but can we just always use the `in_memory` behavior and put a `.load` as the very first transform? Seems unnecessary to have a lazy object (Batches) which contain lazy objects (dask arrays).

Reply:
isel with random indices is only an issue with dask...so maybe just stop using dask?

Reply:
I think what you're proposing (prepending a `load` transform, instead of removing shuffling) would work.

If you go this route, we don't need the "sort by time" transform at the end, this was solely to improve performance.

Reply:
If `shuffle_samples` is True, the shuffle occurs after sorting by time.

`select_random_ordered_subset` is deleted and broken into a short `sort_by_time` function and `select_fraction`. The latter replaces the `select_first_samples` function and has the suggested tests added.